portfolio/src/main/webapp/index.html

```diff
@@ -23,97 +24,98 @@
+                <li>McDiver Sewer Escape</li>
+                <!-- Video demo of game -->
+                <li>Tic Tac Toe Game</li>
+                <!-- Video demonstration -->
+                <li>Hyperharmonic Centrality</li>
+                <!-- Lookup how to convert PDF to image type or how to open pdfs in HTML -->
+            </ul>
+        </div>
+        <div id="recent-activities">
+            <h2>Recent Activities</h2>
+            <!--recently watched, currently watching, will watch -->
+            <!-- I want to figure out how to put these side by side-->
+            <h4>Shows</h4>
+            <img src="images/demon-slayer.jfif" alt="Demon Slayer" style="width:256px;" />
+            <figcaption>Fig.1 - Recently watched (Highly Recommend!)</figcaption>
+            <img src="images/cursed-poster.jpg" alt="Cursed" style="width:256px;" />
+            <figcaption>Fig.2 - Currently watching</figcaption>
+            <img src="images/Attack-On-Titan.jpg" alt="Attack on Titan" style="width:256px;" />
+            <figcaption>Fig.3 - Will watch at some point :)</figcaption>
+            <!-- previous video game, current video game, next game I will play-->
+            <h4>Video Games</h4>
+            <img src="images/BOTW.jpg" alt="Breath of the Wild" style="width:256px;" />
+            <figcaption>Fig.5 - Recently played: Legend of Zelda - Breath of the Wild
+            </figcaption>
+            <img src="images/Minecraft.png" alt="minecraft" style="width:256px;" />
+            <figcaption>Fig.6 - Currently playing: Minecraft (my base)</figcaption>
+            <img src="images/xenoblade-chronicles-2.jpg" alt="Xenoblade" style="width:256px;" />
+            <figcaption>Fig.7 - Will play: Xenoblade Chronicles 2</figcaption>
+        </div>
+        <div id="greeting-container">
+            <h2>A Little Bit About Me</h2>
+            <p style="text-indent: 5%;">My name is Sam and I am a person who
+                enjoys the world of Computer science. I like to design applications
+                to make life more convenient and easier. I want to make the world
+                better through innovative ideas and clever solutions.
+            </p>
+            <p style="text-indent: 5%;">Right now I am a sophomore studying
+                Computer Science with a focus in Machine Learning at Cornell
+                University. Go Big Red!
+            </p>
+            <img src="images/cornell-campus.jpg" alt="Cornell Campus" />
 
-          <!-- It would be cool to make a button that generates random facts
+            <!-- It would be cool to make a button that generates random facts
                 (the things below)-->
+            <p>Click this button to see the server's random fact.</p>
+            <button onclick="showRandText();">
+                Get Random Message
+            </button>
+            <p id="text-container"></p>
+            <p>Click <a href="/rand_response">here</a> to see a random fact.</p>
```

Comment:
I think this functionality is supposed to be in the JS file. When you click the button, make a GET request that gives a random fact by reaching the rand_response endpoint. It seems like you are already doing this, so I don't see the point of having this here.